src/compound_model/CompoundModelFactory.py

```diff
@@ -0,0 +1,403 @@
+            input (dict[str, np.ndarray]): Standardized input representation.
+                Where key = parameter name, and the value = array of values
+                (in float range of [-1,1]) of that parameter. Each value in the
+                array represents one generated image. For example, image 1 should
+                be based on value indexed 1 in the array for each parameter in input.keys().
+                Standardized representation refers to the float range of [-1,1].
+            pop_name (str): Name of the population.
+            append (bool, optional): True if generated samples should be appended to the population.
+                False if samples should replace the population instead. Defaults to True.
+
+        Raises:
+            AssertionError: When the number of latent codes are below 1.
+            AssertionError: When shape of the `latent_codes` is not valid.
+            AssertionError: When a attribute is not supported by the controller.
+        """
+        # Check input
+        assert len(latent_codes.shape) > 1
+        num_samples = latent_codes.shape[0]
+        assert num_samples > 0
+        for attr in input.keys():
+            assert attr in self._controller.get_attributes()
+
+        pop = Population(pop_name)
+
+        # Parse input
+        parsed_input = self._controller.parse_native_input(input)
+
+        # Generate images
+        uris, manipulated_latent_codes = self._controller.generate_native(
+            latent_codes, parsed_input
+        )
+
+        # Add images to population
+        pop.add_all(
+            manipulated_latent_codes,
+            latent_codes,
+            uris,
+            [1] * len(uris),
+            append=append,
+            **input,
+        )
+
+        # Clear compound metrics
+        self._clear_compound_metrics(pop_name)
+
+    def generate_population(
+        self, num_samples: int, num_attrs=-1, replace=False
+    ) -> None:
+        """
+        Generates a random population with specified number of samples, or appends samples to a
+        already existing population.
+
+        The number of attributes manipulated for each sample is specified by number of attributes.
+        (Input sampling to be used by the controller is defined by the controller function `sample_random_input()`.)
+
+        An example how the generated population interacts with pre-existing population:
+
+        Let P be a population with 5 number of samples with id 1,2,3,4,5.
+        'generate_population(num_samples = 15, replace=False)' will result in a population with the prexisting samples
+        1-5 and newly created samples 6-15, if replace would be set to true, id 1-5 would be replaced by new samples.
+
+        Another example:
+
+        Let P be a population with 5 number of samples with id 1,2,3,4,5.
+        'generate_population(num_samples = 3, replace=True)' will result in id 1-3 being replaced by new samples and
+        4-5 staying as the were.
```

Comment:
"as _they_ were, _i.e., the population size would stay the same._"

note "the" -> "they"